### PR TITLE
Modify segment dump to exclude certain metadata.

### PR DESCRIPTION
**Pull request by jimmyyih**
The dump process was checking the segments for several kinds of
metadata, such as indexes and triggers, even though they were not
being included in the dump.  The statements that were performing
these useless checks have been modified to only run on the master.

Authors: James McAtamney and Jimmy Yih

### CODE REVIEW

File: gpMgmt/bin/gppylib/test/behave/mgmt_utils/backup.feature [hunk 1]
```diff
@@ -4868,6 +4868,20 @@ Feature: Validate command line arguments
...
+        And the timestamp from gpcrondump is stored
+        Given database "testdb" is dropped and recreated
+        When the user runs "gpdbrestore -a" with the stored timestamp
+        Then gpdbrestore should return a return code of 0
+        And the user runs "psql -f gppylib/test/behave/mgmt_utils/steps/data/check_metadata.sql testdb > /tmp/check_metadata.out"
+        Then verify that the contents of the files "/tmp/check_metadata.out" and "gppylib/test/behave/mgmt_utils/steps/data/check_metadata.ans" are identical
+
     Scenario: Restore -T for full dump should restore GRANT privileges for tablenames with English and multibyte (chinese) characters
         Given the database is running
         And there are no backup files
```
Comment (tangp3) [inline gpMgmt/bin/gppylib/test/behave/mgmt_utils/backup.feature]: It'd be nice to add tests that verify segments now won't collect certain metadata.

**Comment by hlinnaka** [comment]
I was going to say we should just use the --data-only flag to avoid all that overhead. But looking at the code, I think we still run all the queries to get information on all the objects, even with --data-only. They're just not included in the output. I wonder why that is?

**Comment by jimmyyih** [comment]
I'm not too sure either but we'll probably review the process at a later date.

File: gpAux/extensions/pgbouncer/source [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit 96c8ea1162f96a955ff4d5272ce5bb5fb2ffc44e
+Subproject commit 80ad018f5bcffe01b7aaf96a5ae752fb72e48248
```
Comment (Chibin) [inline gpAux/extensions/pgbouncer/source]: These submodule changes needs to get removed?
Looks like they're reverting to the older commits.

**Comment by jimmyyih** [comment]
Pushed into master after making requested changes from tangp3 and Chibin:
https://github.com/greenplum-db/gpdb/commit/421d8268976d8c3e619f6b9a633feb66759ebf37

**Pull request closed without merging**